File: ontrack-security/src/test/java/net/nemerosa/ontrack/security/EncryptionServiceImplTest.java
package net.nemerosa.ontrack.security;

import org.junit.Before;
import org.junit.Test;

import static org.mockito.Mockito.*;

public class EncryptionServiceImplTest {

    private EncryptionServiceImpl service;
    private ConfidentialKey key;

    @Before
    public void before() {
        key = mock(ConfidentialKey.class);
        service = new EncryptionServiceImpl(key);
    }

    @Test
    public void encrypt() throws Exception {
        service.encrypt("test");
        verify(key, times(1)).encrypt("test");
    }

    @Test
    public void decrypt() throws Exception {
        service.decrypt("xxxx");
        verify(key, times(1)).decrypt("xxxx");
    }

    @Test
    public void encrypt_null() throws Exception {
        service.encrypt(null);
        verify(key, never()).encrypt(anyString());
    }

    @Test
    public void decrypt_null() throws Exception {
        service.decrypt(null);
        verify(key, never()).decrypt(anyString());
    }
}

File: ontrack-service/src/test/java/net/nemerosa/ontrack/service/security/EncryptionServiceImplTest.kt
package net.nemerosa.ontrack.service.security

import io.mockk.mockk
import io.mockk.verify
import net.nemerosa.ontrack.model.security.SecurityService
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class EncryptionServiceImplTest {

    private lateinit var service: EncryptionServiceImpl
    private lateinit var key: ConfidentialKey

    @BeforeEach
    fun before() {
        key = mockk<ConfidentialKey>(relaxed = true)
        val securityService = mockk<SecurityService>(relaxed = true)
        service = EncryptionServiceImpl(securityService, key)
    }

    @Test
    fun encrypt() {
        service.encrypt("test")
        verify(exactly = 1) {
            key.encrypt("test")
        }
    }

    @Test
    fun decrypt() {
        service.decrypt("xxxx")
        verify(exactly = 1) {
            key.decrypt("xxxx")
        }
    }

    @Test
    fun encrypt_null() {
        service.encrypt(null)
        verify(exactly = 0) {
            key.encrypt(any())
        }
    }

    @Test
    fun decrypt_null() {
        service.decrypt(null)
        verify(exactly = 0) {
            key.decrypt(any())
        }
    }
}